packages/memory-storage/src/index.ts

```diff
@@ -0,0 +1,112 @@
+import defaultLog from '@apify/log';
+import type * as storage from '@crawlee/types';
+import { Dictionary } from '@crawlee/utils';
+import { s } from '@sapphire/shapeshift';
+import { existsSync } from 'node:fs';
+import { resolve } from 'node:path';
+import { Worker } from 'node:worker_threads';
+import { DatasetClient } from './resource-clients/dataset';
+import { DatasetCollectionClient } from './resource-clients/dataset-collection';
+import { KeyValueStoreClient } from './resource-clients/key-value-store';
+import { KeyValueStoreCollectionClient } from './resource-clients/key-value-store-collection';
+import { RequestQueueClient } from './resource-clients/request-queue';
+import { RequestQueueCollectionClient } from './resource-clients/request-queue-collection';
+import { WorkerReceivedMessage } from './utils';
+import { FileStorageWorkerEmulator } from './workers/file-storage-emulator';
+
+export interface MemoryStorageOptions {
+    /**
+     * Path to directory where the data will also be saved.
+     * @default process.env.APIFY_LOCAL_STORAGE_DIR ?? './memory-storage'
+     */
+    localDataDirectory?: string;
+}
+
+export class MemoryStorage implements storage.StorageClient {
+    readonly localDataDirectory: string;
+    readonly datasetsDirectory: string;
+    readonly keyValueStoresDirectory: string;
+    readonly requestQueuesDirectory: string;
+
+    readonly keyValueStoresHandled: KeyValueStoreClient[] = []; ;
+    readonly datasetClientsHandled: DatasetClient[] = [];
+    readonly requestQueuesHandled: RequestQueueClient[] = []; ;
+
+    private fileStorageWorker!: Worker | FileStorageWorkerEmulator;
```

Comment:
could the `FileStorageWorkerEmulator` maybe extend `Worker` so we dont have to do this explicitly here?

Reply:
I could make `FileStorageWorkerEmulator` implement the `Worker` interface, but then most of the methods will throw/no-op. I don't know if that's a good idea vs just having this

Reply:
Well technically we only really need a `postMessage` method, so we could just have an interface for that. I'll wait for your answer on this review comment, and will proceed after. 
<img width="808" alt="Google Chrome Dev - 2022-05-17 at 17 56 54@2x" src="https://user-images.githubusercontent.com/17960496/168842520-25935755-511c-448c-8cd7-a33d3bf75138.png">